utils/auth/auth.controller.js

```diff
@@ -0,0 +1,55 @@
+const bcrypt = require("bcryptjs");
+const validate = require("../../resources/users/users.validation");
+const Auth = require("./auth");
+const { genSaltSync, hashSync } = bcrypt;
+
+module.exports = function(model) {
+  return {
+    async createUser(req, res) {
+      const { error } = validate(req.body);
+
+      if (error) {
+        return res
+          .status(422)
+          .json({ status: 422, error: error.details[0].message });
+      }
+      const { first_name, last_name, username, email, password } = req.body;
+      let doc = await model.findOne({ email: req.body.email });
```

Comment:
Since we are validating for uniqueness, you should also check for username `let doc = await model.findOne({ email: req.body.email, username });`